react.md

```diff
@@ -290,18 +290,302 @@ Contains Jest configuration. If you're adding Jest plugins or configuring Enzyme
+```js
+export const MY_ACTION = 'MY_ACTION';
+```
+
+##### Good:
+```js
+export const MY_ACTION = 'myApp/MY_ACTION';
+```
+
+#### Use action creators
+It's tempting to just `dispatch` objects to your store but because we don't like repeating ourselves and we do like writing code that's easy to refactor, we should encapsulate actions in functions.
+
+##### Bad:
+```js
+import { MY_ACTION } from 'state/actions';
+
+dispatch({
+  type: MY_ACTION,
+  myValue,
+});
+```
+
+##### Good:
+```js
+// actions.js
+export const MY_ACTION = 'myApp/MY_ACTION';
+
+export const myAction = myValue => ({
+  type: MY_ACTION,
+  myValue,
+});
+
+// MyComponent/index.js
+import { myAction } from 'state/actions';
+
+dispatch(myAction(myValue));
+```
+
+#### Use a consistent action object format
+`{ type, payload, meta }`
+
+Sticking to a standard action format leads to code that's easier to learn and less surprising. This format is loosely adapted from [Flux Standard Action](https://github.com/redux-utilities/flux-standard-action) but opts to keep errors in `meta` to preserve the expectation that payload is an object of data.
+
+* `type` is always a string action name.
+* `payload` is optional, but when present is data that's meant to be persisted to the store.
+* `meta` is any additional information, like errors, HTTP status, etc. This can be standardized further on a per-application basis.
+
+##### Bad:
+It's not clear from this action what data is persisted to the store and what's just extra metadata.
+
+```js
+// actions.js
+export const MY_ACTION = 'myApp/MY_ACTION';
+
+export const myAction = myValue => ({
+  type: MY_ACTION,
+  myValue,
+  extraDataNotInStore: 'foo',
+});
+```
+
+##### Good:
+Every action uses the same format so other developers can have clear expectations on what's dispatched to reducers. Setting some standards around error formatting, i.e adapting a standard `meta.errors` property, can make it easier to write code like a reducer for global error state handling.
+
+```js
+// actions.js
+export const MY_ACTION = 'myApp/MY_ACTION';
+
+export const myAction = myValue => ({
+  type: MY_ACTION,
+  payload: {
+    myValue,
+  },
+  meta: {
+    extraDataNotInStore: 'foo',
+  }
+});
+```
+
+#### Reducers
+Reducers should:
+
+* Make use of ES6 features like default parameters, destructuring & object spread to reduce boilerplate code.
+* Use `switch` statements to model action branches, not `if/else` object key/value access, or other clever abstractions. While there's a bit more typing involved with `switch`es, it's obvious what they're doing at first glance. Remember, you'll spend more time reading code than writing it.
+
+```js
+import { MY_ACTION } from 'state/actions';
+
+const initialState = {
+  myValue: null,
+  anotherValue: null,
+};
+
+export const myReducer = (state = initialState, { type, payload, meta }) => {
+  switch (type) {
+    case MY_ACTION: {
+      return {
+        // Remember to always destructure the current state first ...
```

Comment:
yes, always this. I might even highlight this - destructing state solves A LOT of common problems that pop up - esp. when we were working on the CoS SFDP